### PR TITLE
Fix connection manager

### PR DESCRIPTION
**Pull request by matthiasbeyer**
Targets #580.

### CODE REVIEW

File: src/connection/manager.c [hunk 1]
```diff
@@ -97,7 +97,7 @@ ws_connection_manager_init(void)
     }
 
     res = ws_socket_init(&connman.sock, create_connection_cb, SOCK_NAME);
-    if (res != 0) {
+    if (res != 0 && res != -EADDRINUSE) {
```
Comment (neithernut) [inline src/connection/manager.c:101]: We shouldn't try to bind to an address twice in the first place. If we do, something else is broken.

**Comment by neithernut** [comment]
I still get console spam about waysome not being able to accept a client, with the code `22` (`EINVAL`).
Might be related to my previous annotation.

**Comment by matthiasbeyer** [comment]
~~Tell me how to reproduce this, please.~~

Just found out...

File: src/connection/manager.c (hunk 1 above)
Comment (matthiasbeyer) [inline src/connection/manager.c:101]: It actually doesn't do this. Remove `/run/user/1000/waysome.sock` after you closed waysome, then try again. Everything is fine then.

Seems that we do not clean up the socket.

**Comment by matthiasbeyer** [comment]
The spamming comes from [here](https://github.com/waysome/waysome/blob/develop/src/util/socket.c#L63), shall we remove this log call (is it intended that this runs `while(true)` like)?

**Comment by neithernut** [comment]
There should be not log message if `accept()` causes `errno` to be `EAGAIN` or `EWOULDBLOCK`.
What do you mean _excactely_ with "`while(true)`-like?".

File: src/connection/manager.c (hunk 1 above)
Comment (neithernut) [inline src/connection/manager.c:101]: Then we should. Or set some flags so bind just takes over the old file.

File: src/util/socket.c [hunk 2]
```diff
@@ -60,7 +60,9 @@ socket_build_connection_cb(
     memset(&addr, 0, len);
     int fd = accept(s->fd, (struct sockaddr*) &addr, &len);
     if (fd < 0) {
-        ws_log(&log_ctx, LOG_ERR, "Could not accept client: %d", errno);
+        if (fd != -EAGAIN && fd != -EWOULDBLOCK) {
+            ws_log(&log_ctx, LOG_ERR, "Could not accept client: %d", errno);
+        }
```
Comment (neithernut) [inline src/util/socket.c:65]: You can merge the `if`s.

Comment (matthiasbeyer) [inline src/util/socket.c:65]: nope. re-read the patch!

**Comment by matthiasbeyer** [comment]
Weird. After removing the file, I get my terminal spammed by the messages you mentioned, because `accept()` returns `-EINVAL`. Running waysome a second time (so, the file already exists), this error doesn't happen again.

Any idea why this happenes?

**Comment by neithernut** [comment]
Note for anybody reviewing this: we did not `listen()`.

**Comment by neithernut** [comment]
ACK for squashing (if there's anything to squash)

**Comment by neithernut** [comment]
ACK.

**Pull request merged**